tools/cldr-code/src/main/java/org/unicode/cldr/test/ExampleGenerator.java

```diff
@@ -769,16 +776,39 @@ private String handleFormatUnit(XPathParts parts, String value) {
                 }
                 String minimalPattern = cldrFile.getStringValue("//ldml/numbers/minimalPairs/caseMinimalPairs[@case=\"" + gCase + "\"]");
                 if (minimalPattern != null && numberFormat != null) {
-                    String composed = format(value, backgroundStartSymbol + numberFormat.format(amount) + backgroundEndSymbol);
+                    String composed = format(unitPattern, backgroundStartSymbol + formattedAmount + backgroundEndSymbol);
                     examples.add(backgroundStartSymbol + format(minimalPattern, backgroundEndSymbol + composed + backgroundStartSymbol) + backgroundEndSymbol);
-                } else if (unitCaseInfo != null && !unitCaseInfo.isEmpty()) {
-                    examples.add("⚠️No Case Minimal Pair available yet️");
+                    // get contrasting case
+                    if (unitCaseInfo != null && !unitCaseInfo.isEmpty()) {
+                        String constrastingCase = getConstrastingCase(unitPattern, gCase, unitCaseInfo, parts);
+                        if (constrastingCase != null) {
+                            minimalPattern = cldrFile.getStringValue("//ldml/numbers/minimalPairs/caseMinimalPairs[@case=\"" + constrastingCase + "\"]");
+                            composed = format(unitPattern, backgroundStartSymbol + formattedAmount + backgroundEndSymbol);
+                            examples.add(EXAMPLE_OF_INCORRECT + backgroundStartSymbol + format(minimalPattern, backgroundEndSymbol + composed + backgroundStartSymbol) + backgroundEndSymbol);
+                        }
+                    } else {
+                        examples.add(EXAMPLE_OF_CAUTION + "️No Case Minimal Pair available yet️");
+                    }
                 }
             }
         }
         return formatExampleList(examples);
     }
 
+    private String getConstrastingCase(String unitPattern, String gCase, Collection<String> unitCaseInfo, XPathParts parts) {
+        for (String otherCase : unitCaseInfo) {
+            if (otherCase.equals(gCase)) {
+                continue;
+            }
+            parts.putAttributeValue(-1, "case", "nominative".equals(otherCase) ? null : otherCase);
```

Comment:
could constants be used instead of literal "case" and "nominative"? maybe enums rather than just strings?

Reply:
Both good comments. Will handle in followup pr